contracts/zero-ex/contracts/src/features/MultiplexFeature.sol

```diff
@@ -0,0 +1,799 @@
+                    //     into the FlashWallet, or
+                    // (2) args.inputToken == ETH_TOKEN_ADDRESS, in which
+                    //     case ETH is attached to the call and no token
+                    //     transfer occurs.
+                    args.inputTokenAmount = outputTokenAmount;
+                }
+                uint256 ethValue;
+                (args.transformations, ethValue) = abi.decode(
+                    wrappedCall.data,
+                    (ITransformERC20Feature.Transformation[], uint256)
+                );
+                // Do not spend more than the remaining ETH.
+                ethValue = LibSafeMathV06.min256(ethValue, remainingEth);
+                // Call `_transformERC20`.
+                outputTokenAmount = ITransformERC20Feature(address(this))
+                    ._transformERC20{value: ethValue}(args);
+                // Decrement the remaining ETH.
+                remainingEth -= ethValue;
+                // If the recipient was not `msg.sender`, it must be the target
+                // contract for the next hop.
+                nextTarget = args.taker == msg.sender ? address(0) : args.taker;
+            } else if (wrappedCall.selector == this._batchFill.selector) {
+                BatchFillData memory batchFillData;
+                batchFillData.inputToken = IERC20TokenV06(fillData.tokens[i]);
+                batchFillData.outputToken = IERC20TokenV06(fillData.tokens[i + 1]);
+                uint256 ethValue;
+                (batchFillData.calls, ethValue) = abi.decode(
+                    wrappedCall.data,
+                    (WrappedBatchCall[], uint256)
+                );
+                // Do not spend more than the remaining ETH.
+                ethValue = LibSafeMathV06.min256(ethValue, remainingEth);
+                if (IERC20TokenV06(fillData.tokens[i]).isTokenETH()) {
+                    batchFillData.sellAmount = LibSafeMathV06.min256(
+                        outputTokenAmount,
+                        ethValue
+                    );
+                } else {
+                    batchFillData.sellAmount = outputTokenAmount;
+                }
+                // Decrement the remaining ETH.
+                remainingEth -= ethValue;
+                uint256 leftoverEth;
+                (outputTokenAmount, leftoverEth) = _batchFill(batchFillData, ethValue);
+                // Add back any ETH that was unused by the `_batchFill`.
+                remainingEth += leftoverEth;
+                nextTarget = address(0);
+            } else if (wrappedCall.selector == IEtherTokenV06.deposit.selector) {
+                uint256 ethValue = LibSafeMathV06.min256(outputTokenAmount, remainingEth);
+                // Wrap ETH.
+                weth.deposit{value: ethValue}();
+                nextTarget = _computeHopRecipient(fillData.calls, i);
+                weth.transfer(nextTarget, ethValue);
```

Comment:
Can you think of any risks that may arise if `nextTarget` re-enters the EP?

Reply:
Oh, as well is this the standard now?
```
(bool success,) = msg
                .sender
                .call{value: ethValue}("");
```

Reply:
`nextTarget` can only be set to one of the following:
- address(0)
- msg.sender
- a Uniswap/Sushiswap pair contract
- a LiquidityProvider contract
- the FlashWallet
- the EP
The only case in which `nextTarget` is called directly is if it's a Uniswap or Sushiswap pair contract, which is computed using their respective factories and init code hashes. If it's a user-supplied LiquidityProvider contract, it's called using the sandbox as a proxy. It's only set to the EP if the next hop is a WETH unwrap.